pallets/bridge/src/lib.rs

```diff
@@ -393,16 +394,47 @@ decl_event! {
 }
 
 decl_module! {
-    pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+    pub struct Module<T: Trait> for enum Call where origin: <T as frame_system::Trait>::Origin {
         type Error = Error<T>;
 
-        const MaxTimelockedTxsPerBlock: u32 = T::MaxTimelockedTxsPerBlock::get();
-
         fn deposit_event() = default;
 
-        /// Issues tokens in timelocked transactions.
-        fn on_initialize(block_number: T::BlockNumber) -> Weight {
-            Self::handle_timelocked_txs(block_number)
+        fn on_runtime_upgrade() -> Weight {
+            use frame_support::{migration::StorageKeyIterator, Twox64Concat};
+
+            let now = frame_system::Module::<T>::block_number();
+
+            // Migrate timelocked transactions.
+            StorageKeyIterator::<T::BlockNumber, Vec::<BridgeTx<T::AccountId, T::Balance>>, Twox64Concat>::new(b"Bridge", b"TimelockedTxs")
+                .drain()
+                .filter_map(|(block_number, txs)| {
+                    // Schedule only for future blocks.
+                    let block_number = T::BlockNumber::max(block_number, now + One::one());
+                    let calls: Vec<_> = txs
+                        .into_iter()
+                        .map(|tx| {
+                            Call::<T>::handle_scheduled_bridge_tx(tx).into()
+                        })
+                        .collect();
```

Comment:
Although this is a one-time-code thing, consider a `.flat_map(...)` instead for efficiency (don't allocate into a `Vec<_>` that is consumed immediately).

Reply:
I've changed that to a `.map` without collecting into a `Vec`. Please check if that's OK with you.